hbase_sql/orc-spark-3.0/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java

```diff
@@ -0,0 +1,87 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.parquet.hadoop.ParquetOutputFormat;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder;
+import org.apache.spark.sql.execution.datasources.parquet.ParquetWriteSupport;
+import org.apache.spark.sql.internal.SQLConf;
+import org.apache.spark.sql.types.StructType;
+
+import java.io.IOException;
+
+public class ParquetWriterFactoryImpl implements ParquetWriterFactory{
+    
+    @Override
+    public RecordWriter<Void, Object> getParquetRecordWriter(String location, String compression, StructType tableSchema) throws IOException, InterruptedException {
+        ParquetWriteSupport pws = new ParquetWriteSupport();
+        final Configuration conf = new Configuration((Configuration) EngineDriver.driver().getConfiguration().getConfigSource().unwrapDelegate());
+        conf.set(SQLConf.PARQUET_WRITE_LEGACY_FORMAT().key(), "false");
+        conf.set(SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE().key(), "INT96");
+        conf.set(SQLConf.PARQUET_INT96_AS_TIMESTAMP().key(), "true");
+        conf.set(SQLConf.PARQUET_BINARY_AS_STRING().key(), "false");
+
+        pws.setSchema(tableSchema, conf);
+        return new ParquetOutputFormat(new ParquetWriteSupport()) {
+            @Override
+            public Path getDefaultWorkFile(TaskAttemptContext context, String extension) throws IOException {
+                return new Path(location+"/part-r-00000"+extension);
+            }
+
+            @Override
+            public RecordWriter<Void, Object> getRecordWriter(TaskAttemptContext taskAttemptContext)
+                    throws IOException, InterruptedException {
+
+                CompressionCodecName codec;
+                switch (compression) {
+                    case "none":
+                        codec = CompressionCodecName.UNCOMPRESSED;
+                        break;
+                    case "snappy":
+                        codec = CompressionCodecName.SNAPPY;
+                        break;
+                    case "lzo":
+                        codec = CompressionCodecName.LZO;
+                        break;
+                    case "gzip":
+                    case "zip":
+                        codec = CompressionCodecName.GZIP;
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown compression: " + compression);
+                }
+                String extension = codec.getExtension() + ".parquet";
+                Path file = getDefaultWorkFile(taskAttemptContext, extension);
+                return getRecordWriter(conf, file, codec);
+            }
+        }.getRecordWriter(null);
```

Comment:
for other reviewers: the difference between this file and e.g. `hbase_sql/orc-spark-2.4/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java` is

```
<         return encoder.createSerializer().apply(valueRow);
---
>         return encoder.toRow(valueRow);
```